app/controllers/steps/case/charges_controller.rb

```diff
@@ -13,6 +13,27 @@ def update
         )
       end
 
+      def destroy
+        charge_record.destroy
+
+        if case_charges.reload.any?
+          redirect_to edit_steps_case_charges_summary_path,
+                      success: t('.success_flash')
+        else
+          # If this was the last remaining record, redirect
+          # to the charges page with a new blank one
+          charge = case_charges.create!(
```

Comment:
There is a bit of repetition here and in the decision tree. For now I've not addressed this as is very minor, but if we see we have to use this in more places, it may be we could expose a method on the relationship named `create_blank!` or something.

Reply:
I've already cleanup the duplication and extracted the creation of the blank associated date to `Case` has_many relationship in a recent commit 😄